sbi/inference/base.py

```diff
@@ -306,7 +306,7 @@ def train(
         discard_prior_samples: bool = False,
         retrain_from_scratch: bool = False,
         show_train_summary: bool = False,
-    ) -> NeuralPosterior:
+    ) -> NeuralPotentialPosterior:
```

Comment:
Since `NSPE` also inherits from `NeuralInference`, this should still return a `NeuralPosterior`